internal/portforward/service/service.go

```diff
@@ -47,3 +47,45 @@ func (s *Service) GetPortsForwarded() (ports []uint16) {
 	copy(ports, s.ports)
 	return ports
 }
+
+func (s *Service) SetPortsForwarded(ctx context.Context, ports []uint16) (err error) {
+	for i, port := range s.ports {
+		err := s.portAllower.RemoveAllowedPort(ctx, port)
+		if err != nil {
+			for j := 0; j < i; j++ {
```

Comment:
nit you can now use the 'intrange' introduced in Go 1.23

```suggestion
			for j := range i {
```

Reply:
"Fixed" with a1e7f128d97494966c99d457b40cfcf90ff0c198